commands/plugin_marketplace_e2e_test.go

```diff
@@ -0,0 +1,143 @@
+	"github.com/mattermost/mmctl/printer"
+)
+
+func (s *MmctlE2ETestSuite) TestPluginMarketplaceInstallCmd() {
+	s.SetupTestHelper().InitBasic()
+
+	s.RunForSystemAdminAndLocal("install a plugin", func(c client.Client) {
+		printer.Clean()
+
+		const (
+			pluginID      = "jira"
+			pluginVersion = "3.0.0"
+		)
+
+		defer removePluginIfInstalled(s, pluginID)
+
+		err := pluginMarketplaceInstallCmdF(c, &cobra.Command{}, []string{pluginID, pluginVersion})
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetErrorLines(), 0)
+		s.Require().Len(printer.GetLines(), 1)
+
+		manifest := printer.GetLines()[0].(*model.Manifest)
+		s.Require().Equal(pluginID, manifest.Id)
+		s.Require().Equal(pluginVersion, manifest.Version)
+
+		plugins, appErr := s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Len(plugins.Active, 0)
+		s.Require().Len(plugins.Inactive, 1)
+		s.Require().Equal(pluginID, plugins.Inactive[0].Id)
+		s.Require().Equal(pluginVersion, plugins.Inactive[0].Version)
+	})
+
+	s.Run("install a plugin without permissions", func() {
+		printer.Clean()
+
+		const (
+			pluginID      = "jira"
+			pluginVersion = "3.0.0"
+		)
+
+		defer removePluginIfInstalled(s, pluginID)
+
+		err := pluginMarketplaceInstallCmdF(s.th.Client, &cobra.Command{}, []string{pluginID, pluginVersion})
+		s.Require().NotNil(err)
+		s.Require().Contains(err.Error(), "You do not have the appropriate permissions.")
+		s.Require().Len(printer.GetErrorLines(), 0)
+		s.Require().Len(printer.GetLines(), 0)
+
+		plugins, appErr := s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Len(plugins.Active, 0)
+		s.Require().Len(plugins.Inactive, 0)
+	})
+
+	s.RunForSystemAdminAndLocal("install a plugin without version", func(c client.Client) {
+		printer.Clean()
+
+		const (
+			pluginID = "jira"
+		)
+
+		defer removePluginIfInstalled(s, pluginID)
+
+		err := pluginMarketplaceInstallCmdF(c, &cobra.Command{}, []string{pluginID})
+		s.Require().Nil(err)
+		s.Require().Len(printer.GetErrorLines(), 0)
+		s.Require().Len(printer.GetLines(), 1)
+
+		manifest := printer.GetLines()[0].(*model.Manifest)
+		s.Require().Equal(pluginID, manifest.Id)
+		s.Require().NotEmpty(manifest.Version)
+
+		plugins, appErr := s.th.App.GetPlugins()
+		s.Require().Nil(appErr)
+		s.Require().Len(plugins.Active, 0)
+		s.Require().Len(plugins.Inactive, 1)
+		s.Require().Equal(pluginID, plugins.Inactive[0].Id)
+		s.Require().NotEmpty(plugins.Inactive[0].Version)
+	})
+
+	s.RunForSystemAdminAndLocal("install a plugin with invalid version", func(c client.Client) {
+		printer.Clean()
+
+		const (
+			pluginID      = "jira"
+			pluginVersion = "invalid-version"
+		)
+
+		defer removePluginIfInstalled(s, pluginID)
+
+		err := pluginMarketplaceInstallCmdF(c, &cobra.Command{}, []string{pluginID, pluginVersion})
+		s.Require().NotNil(err)
+		s.Require().Contains(err.Error(), "Could not find the requested marketplace plugin.")
+		s.Require().Len(printer.GetErrorLines(), 0)
```

Comment:
@mgdelacroix its weird to me that `pluginMarketplaceInstallCmdF` just returns the error instead of printing it as well. Commands in `team.go`, `bot.go` and `user.go` print the error along with (or instead of) returning the error. 

Maybe we should fix this inconsistency?

Reply:
We're using both approaches depending on the case (sometimes if there is only one error for one operation, we just return it). One of the changes part of the `mmctl improvements` is to get rid of this inconsistency and basically print everything that is not a flag validation error, so I would suggest to tackle it then